lib/spack/spack/installer.py

```diff
@@ -1702,6 +1704,29 @@ def _install_task(self, task: BuildTask, install_status: InstallStatus) -> None:
             # If a compiler, ensure it is added to the configuration
             if task.compiler:
                 self._add_compiler_package_to_config(pkg)
+
+            # Iterate over all mirrors with autopush==True
+            all_mirrors = spack.mirror.MirrorCollection(binary=True)
+            autopush_mirrors = [mirror for mirror in all_mirrors.values() if mirror.get_autopush()]
+            for mirror in autopush_mirrors:
+                # Create and execute buildcache push command
+                bc_parser = argparse.ArgumentParser()
+                bc.setup_parser(bc_parser)
+                bc_flags = []
+                # The dependencies should already be in the buildcache
+                # No need to add them again (especially with --force)
+                bc_flags.append("--only=package")
+                # We want the pushed packages to be available right away
+                bc_flags.append("--update-index")
```

Comment:
`--update-index` and `--force` probably should no be in there. 

`--update-index` is expensive, not necessarily required, and this also should probably consider updating the signing key index. I think two things for this could be to
  a) move the update index part of this to the post install step that loops the `autopush` mirrors after installing all of the packages
  b) make updating the index an option that is passed.

`--force` just seem strange here. The reason in the comments is not possible, if there is a corrupted item in the build cache Spack does not fall back to installing from source, it simply fails the install.